m4opt/missions/_ztf/__init__.py

```diff
@@ -0,0 +1,187 @@
+    ns_chip_index, ew_chip_index = np.divmod(chipid, ew_nchips)
+    ns_rc_in_chip_index = np.where(rc_in_chip_id <= 1, 1, 0)
+    ew_rc_in_chip_index = np.where((rc_in_chip_id == 0) | (rc_in_chip_id == 3), 0, 1)
+
+    ew_offsets = (
+        ew_chip_gap * (ew_chip_index - (ew_nchips - 1) / 2)
+        + ew_npix * plate_scale * (ew_chip_index - ew_nchips / 2)
+        + 0.5 * ew_rc_in_chip_index * plate_scale * ew_npix
+    )
+    ns_offsets = (
+        ns_chip_gap * (ns_chip_index - (ns_nchips - 1) / 2)
+        + ns_npix * plate_scale * (ns_chip_index - ns_nchips / 2)
+        + 0.5 * ns_rc_in_chip_index * plate_scale * ns_npix
+    )
+
+    ew_ccd_corners = 0.5 * plate_scale * np.asarray([ew_npix, 0, 0, ew_npix])
+    ns_ccd_corners = 0.5 * plate_scale * np.asarray([ns_npix, ns_npix, 0, 0])
+
+    coords = SkyCoord(
+        ew_offsets[:, np.newaxis] + ew_ccd_corners[np.newaxis, :],
+        ns_offsets[:, np.newaxis] + ns_ccd_corners[np.newaxis, :],
+    )
+    return Regions([PolygonSkyRegion(coord) for coord in coords])
+
+
+def _read_skygrid():
+    table = Table.read(
+        resources.files(data) / "ZTF_Fields.txt",
+        format="ascii.fixed_width_no_header",
+        delimiter=" ",
+        comment="%",
+    )
+    return SkyCoord(table["col2"], table["col3"], unit=u.deg)
+
+
+ztf = Mission(
+    name="ztf",
+    fov=_make_fov(),
+    constraints=[
+        AirmassConstraint(2.5),
+        AtNightConstraint.twilight_astronomical(),
+        MoonSeparationConstraint(25 * u.deg),
+        # The rest of these are positional constraints from
+        # https://github.com/ZwickyTransientFacility/ztf_sim/blob/5176ebaa9e1f8e5448593df4102d077c0e880886/ztf_sim/QueueManager.py#L1235-L1255
+        HourAngleConstraint(-5.95 * u.hourangle, 5.95 * u.hourangle),
+        LogicalOrConstraint(
+            HourAngleConstraint(-17.6 * u.deg, 180 * u.deg),
+            DeclinationConstraint(-22 * u.deg, 90 * u.deg),
+        ),
+        LogicalOrConstraint(
+            HourAngleConstraint(-180 * u.deg, -17.6 * u.deg),
+            DeclinationConstraint(-45 * u.deg, 90 * u.deg),
+        ),
+        LogicalOrConstraint(
+            HourAngleConstraint(-3 * u.deg, 3 * u.deg),
+            DeclinationConstraint(-46 * u.deg, 90 * u.deg),
+        ),
+        DeclinationConstraint(-90 * u.deg, 87.5 * u.deg),
+    ],
+    detector=Detector(
+        # Table 1 of https://ui.adsabs.harvard.edu/abs/2020PASP..132c8001D
+        area=np.pi * np.square(0.5 * 1244.6 * u.mm),
+        # Table 3 of https://ui.adsabs.harvard.edu/abs/2020PASP..132c8001D
+        plate_scale=(1.01 * u.arcsec) ** 2,
+        read_noise=8,
+        gain=6.2,
+        bandpasses={band: bandpass_from_svo(f"Palomar/ZTF.{band}") for band in "gri"},
+        background=SkyBackground.medium() + ZodiacalBackground(),
+    ),
+    observer_location=EarthFixedObserverLocation.of_site("Palomar"),
+    skygrid=_read_skygrid(),
+    # From Section 4.2:
```

Comment:
We have a more detailed model available [here](https://github.com/ZwickyTransientFacility/ztf_sim/blob/5176ebaa9e1f8e5448593df4102d077c0e880886/ztf_sim/constants.py#L27C1-L36C45) although I personally haven't validated it against the last N years of on-sky data taking.

Reply:
OK, it's easy enough for us to model that using separate "bang-bang" models for each Euler axis.

Does ZTF have a "dome crawl" feature like LSST, as @rhiannonlynne described in https://github.com/m4opt/m4opt/pull/292#discussion_r1997455512?